core/src/com/unciv/logic/city/CityStats.kt

```diff
@@ -174,15 +174,27 @@ class CityStats {
         if (!civInfo.isPlayerCivilization())
             unhappinessModifier *= civInfo.gameInfo.getDifficulty().aiUnhappinessModifier
 
-        var unhappinessFromCity = -3f     // -3 happiness per city
+        var unhappinessFromCity = -3f // -3 happiness per city
         if (civInfo.hasUnique("Unhappiness from number of Cities doubled"))
-            unhappinessFromCity *= 2f//doubled for the Indian
+            unhappinessFromCity *= 2f //doubled for the Indian
 
         newHappinessList["Cities"] = unhappinessFromCity * unhappinessModifier
 
         var unhappinessFromCitizens = cityInfo.population.population.toFloat()
-        if (civInfo.hasUnique("Specialists produce half normal unhappiness"))
-            unhappinessFromCitizens -= cityInfo.population.getNumberOfSpecialists() * 0.5f
+        var unhappinessFromSpecialists = cityInfo.population.getNumberOfSpecialists().toFloat()
+        
+        unhappinessFromCitizens -= unhappinessFromSpecialists
+        
+        
+        for (unique in civInfo.getMatchingUniques("Specialists only produce []% of normal unhappiness")) {
+            unhappinessFromSpecialists *= (1f - unique.params[0].toFloat() / 100f)
+        }
+        // Deprecated since 3.15
+            if (civInfo.hasUnique("Specialists produce half normal unhappiness"))
+                unhappinessFromSpecialists *= 0.5f
+        //
+        
+        unhappinessFromCitizens += unhappinessFromSpecialists
```

Comment:
Surely it would make more sense to just remove the unhappiness at the end instead of adding and then subtracting?

Reply:
It would, but then I would have to do maths and this was easier :p
I'll update it in a bit.